Still present random `SIGSEGV segmentation violation` in `action-metrics-server`

### Checks

- [X] I've already read https://github.com/actions/actions-runner-controller/blob/master/TROUBLESHOOTING.md and I'm sure my issue is not covered in the troubleshooting guide.
- [X] I'm not using a custom entrypoint in my runner image

### Controller Version

0.27.1

### Helm Chart Version

0.22.0

### CertManager Version

v1.11.0

### Deployment Method

Helm

### cert-manager installation

Installet correctly

### Checks

- [X] This isn't a question or user support case (For Q&A and community support, go to [Discussions](https://github.com/actions/actions-runner-controller/discussions). It might also be a good idea to contract with any of contributors and maintainers if your business is so critical and therefore you need priority support
- [X] I've read [releasenotes](https://github.com/actions/actions-runner-controller/tree/master/docs/releasenotes) before submitting this issue and I'm sure it's not due to any recently-introduced backward-incompatible changes
- [X] My actions-runner-controller version (v0.x.y) does support the feature
- [X] I've already upgraded ARC (including the CRDs, see charts/actions-runner-controller/docs/UPGRADING.md for details) to the latest and it didn't fix the issue
- [X] I've migrated to the workflow job webhook event (if you using webhook driven scaling)

### Resource Definitions

```yaml
image:
  tag: v0.27.1
  pullPolicy: Always
actionsMetricsServer:
  enabled: true
```


### To Reproduce

```markdown
Run `actionsMetricsServer` with a latest image
```


### Describe the bug

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1420079]

goroutine 15 [running]:
github.com/actions/actions-runner-controller/pkg/actionsmetrics.(*EventReader).fetchAndParseWorkflowJobLogs(0x1583e40?, {0x19d0ed0?, 0xc0001a2000?}, 0xe?)
	github.com/actions/actions-runner-controller/pkg/actionsmetrics/event_reader.go:208 +0x59
github.com/actions/actions-runner-controller/pkg/actionsmetrics.(*EventReader).ProcessWorkflowJobEvent(0xc00018abd0, {0x19d0ed0, 0xc0001a2000}, {0x15fa880?, 0xc0005f01b0})
	github.com/actions/actions-runner-controller/pkg/actionsmetrics/event_reader.go:139 +0x105b
github.com/actions/actions-runner-controller/pkg/actionsmetrics.(*EventReader).ProcessWorkflowJobEvents(0xc00018abd0, {0x19d0ed0, 0xc0001a2000})
	github.com/actions/actions-runner-controller/pkg/actionsmetrics/event_reader.go:44 +0x59
main.main.func1()
	github.com/actions/actions-runner-controller/cmd/actionsmetricsserver/main.go:157 +0x6c
created by main.main
	github.com/actions/actions-runner-controller/cmd/actionsmetricsserver/main.go:154 +0xdef
```

### Describe the expected behavior

See the metrics

### Whole Controller Logs

```shell
Controller logs is fine
```


### Whole Runner Pod Logs

```shell
Runner logs is fine
```


### Additional Context

Related issue: https://github.com/actions/actions-runner-controller/issues/2309

--- a/pkg/actionsmetrics/event_reader.go
+++ b/pkg/actionsmetrics/event_reader.go
@@ -136,12 +136,27 @@ func (reader *EventReader) ProcessWorkflowJobEvent(ctx context.Context, event in
 		// job_conclusion -> (neutral, success, skipped, cancelled, timed_out, action_required, failure)
 		githubWorkflowJobConclusionsTotal.With(extraLabel("job_conclusion", *e.WorkflowJob.Conclusion, labels)).Inc()
 
-		parseResult, err := reader.fetchAndParseWorkflowJobLogs(ctx, e)
-		if err != nil {
-			log.Error(err, "reading workflow job log")
-			return
-		} else {
-			log.Info("reading workflow_job logs", keysAndValues...)
+		var (
+			exitCode       = "na"
+			runTimeSeconds *float64
+		)
+
+		// We need to do our best not to fail the whole event processing
+		// when the user provided no GitHub API credentials.
+		// See https://github.com/actions/actions-runner-controller/issues/2424
+		if reader.GitHubClient != nil {
+			parseResult, err := reader.fetchAndParseWorkflowJobLogs(ctx, e)
+			if err != nil {
+				log.Error(err, "reading workflow job log")
+				return
+			}
+
+			exitCode = parseResult.ExitCode
+
+			s := parseResult.RunTime.Seconds()
+			runTimeSeconds = &s
+
+			log.WithValues(keysAndValues...).Info("reading workflow_job logs", "exit_code", exitCode)
 		}
 
 		if *e.WorkflowJob.Conclusion == "failure" {
@@ -167,18 +182,20 @@ func (reader *EventReader) ProcessWorkflowJobEvent(ctx context.Context, event in
 				}
 				if *conclusion == "timed_out" {
 					failedStep = fmt.Sprint(i)
-					parseResult.ExitCode = "timed_out"
+					exitCode = "timed_out"
 					break
 				}
 			}
 			githubWorkflowJobFailuresTotal.With(
 				extraLabel("failed_step", failedStep,
-					extraLabel("exit_code", parseResult.ExitCode, labels),
+					extraLabel("exit_code", exitCode, labels),
 				),
 			).Inc()
 		}
 
-		githubWorkflowJobRunDurationSeconds.With(extraLabel("job_conclusion", *e.WorkflowJob.Conclusion, labels)).Observe(parseResult.RunTime.Seconds())
+		if runTimeSeconds != nil {
+			githubWorkflowJobRunDurationSeconds.With(extraLabel("job_conclusion", *e.WorkflowJob.Conclusion, labels)).Observe(*runTimeSeconds)
+		}
 	}
 }
 

